packages/integration-sdk-testing/src/jest.ts

```diff
@@ -536,7 +536,24 @@ export function toMatchDirectRelationshipSchema<T extends ExplicitRelationship>(
     newRelationshipSchema.properties._type = { const: params._type };
   }
 
-  return toMatchSchema(received, newRelationshipSchema);
+  return toMatchSchema(
+    removeUndefinedProperties(received),
+    newRelationshipSchema,
+  );
+}
+
+function removeUndefinedProperties<T extends Entity | ExplicitRelationship>(
+  received: T | T[],
+) {
+  if (!Array.isArray(received)) {
+    received = [received];
+  }
+  return received.map((r) => {
+    for (const [key, value] of Object.entries(r)) {
+      if (value === undefined) delete r[key];
```

Comment:
Should we be mutating the original object or perhaps we should create a new object and copy over _defined_ values?